cinema/serializers.py

```diff
@@ -78,3 +89,52 @@
     class Meta:
         model = MovieSession
         fields = ("id", "show_time", "movie", "cinema_hall")
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation["taken_places"] = [
+            {"row": ticket.row, "seat": ticket.seat}
+            for ticket in instance.tickets.all()
+        ]
+        return representation
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "movie_session", "row", "seat")
+
+    def validate(self, data):
+        data = super(TicketSerializer, self).validate(data)
+
+        ticket = Ticket(
+            movie_session=data.get("movie_session"),
+            row=data.get("row"),
+            seat=data.get("seat"),
+        )
+        ticket.full_clean()
```

Comment:
Using `full_clean()` on a model instance within a serializer's `validate` method is not typical. This method is usually used to validate model instances before saving them to the database. Consider using serializer validation methods to perform necessary checks, or ensure that `full_clean()` is appropriate for your use case.